source/flow.rst

```diff
@@ -10,4 +10,303 @@ and events generated by devices.
+  be the end-point of the flow. *Specific device* any chosen device wil be the output of the flow and *a device 
+  defined during the flow* will make a device that the flow selected during the execution the endpoint.
+- **Source** *(required)*: Data structure that will be mapped as message to device out
+
+Actuate
+*******
+
+.. _actuate_node:
+.. image:: images/nodes/actuate_node.png
+    :width: 20%
+    :align: left
+    :alt: actuate_node
+
+Actuate node is, basically, the same thing of **device out** node. But, it can send messages
+to a real device, like telling a lamp to turn the light off and etc...
+
+.. _actuate_node_cfg:
+.. figure:: images/nodes/actuate_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: actuate_node_cfg
+
+    : Actuate configuration
+
+Fields:
+
+* **Name** *(optional)*: Name of the node.
+* **Device** *(required)*: A real device on dojot
+* **Source** *(required)*: Data structure that will be mapped as message to device out
+
+Change
+*******
+
+.. _change_node:
+.. image:: images/nodes/change_node.png
+    :width: 20%
+    :align: left
+    :alt: change_node
+
+Change node is used to copy or assign values to an output, i. e., copy
+values of a message attributes to a dictionary that will be assigned to
+virtual device                                                                                      
+                                                                                
+
+.. _change_node_cfg:
+.. figure:: images/nodes/change_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: change_node_cfg
+
+    : Change configuration
+
+Fields:
+
+* **Name** *(optional)*: Name of the node
+* **msg** *(required)*: Definition of the data structure that will be sent to the next node and will
+  receive the value set on the *to* field 
+* **to** *(required)*: Assignment or copy of values
+
+.. note::
+    More than one rule can be assign by clicking on *+add* below the rules box.
+
+Switch
+*******
+
+.. _switch_node:
+.. image:: images/nodes/switch_node.png
+    :width: 20%
+    :align: left
+    :alt: switch_node
+
+The Switch node allows messages to be routed to different branches of a flow by evaluating a set of rules against each message.
+
+.. _switch_node_cfg:
+.. figure:: images/nodes/switch_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: switch_node_cfg
+
+    : Switch configuration
+
+Fields:
+
+* **Name** *(optional)*: Name of the node
+* **Property** *(required)*: Variable that will be evaluated 
+* **Rule box** *(required)*: Rules that will determine the outputch branch of the node.
+  Also, it can be configured to stop checking rules when it finds one that matches other
+  or check all the rules and route the message to the corresponding output.
+
+.. note::
+    - More than one rule can be assign by clicking on *+add* below the rules box.
+    - Each condition should have an corresponding output.
```

Comment:
Here, I think that it would be nice to indicate that the rules are mapped one-to-one to the output connectors, instead of saying that 'rules should have an output'. It seems that the user must do something to be sure that each rule has an output.